src/nanotron/fp8/meta.py

```diff
@@ -0,0 +1,62 @@
+from dataclasses import dataclass
+from typing import Union
+
+import torch
+import transformer_engine as te  # noqa
+import transformer_engine_extensions as tex
+
+from nanotron.fp8.constants import DTYPE_TO_FP8_MAX
+
+
+@dataclass
+class FP8Meta:
+    """Metadata for FP8Tensor."""
+
+    amax: Union[int, float]
+
+    # TODO(xrsrke): change to Literal[torch.int8, torch.uint8]
+    dtype: torch.dtype
+    inverse_scale: float = 1.0
+
+    @property
+    def te_dtype(self) -> tex.DType:
+        from nanotron.fp8.tensor import convert_torch_dtype_to_te_dtype
+
+        return convert_torch_dtype_to_te_dtype(self.dtype)
+
+    def __post_init__(self):
+        # NOTE: transformer engine only accepts torch tensors
+        # NOTE: remove fill scale factor when initializing
+        # TODO(xrsrke): check if we haven't computed the scaling factor, then do an initial scaling factor
+        # don't set the initial to 1
+        self.scale = 1  # initial scaling factor
+        self.scale = self._compute_scaling_factor()
+        self.amax = torch.tensor(self.amax, device="cuda") if not isinstance(self.amax, torch.Tensor) else self.amax
+
+    @property
+    def fp8_max(self) -> float:
+        """Return the maximum normal value for the current dtype."""
+        return DTYPE_TO_FP8_MAX[self.dtype]
+
+    def _compute_scaling_factor(self, margin: float = 0) -> torch.Tensor:
+        # Credits: https://github.com/Azure/MS-AMP/blob/d562f0f0bcfc9b712fa0726b73428753ff1300ab/msamp/common/tensor/meta.py#L39
+        # NOTE: seems like python 3.11.5 don't call __post_init__ when using dataclass
+        # so it don't cast [int, float] to torch.Tensor
+        # so we need to manually cast it
+        amax = torch.tensor(self.amax) if not isinstance(self.amax, torch.Tensor) else self.amax
+        scale = torch.tensor(self.scale) if not isinstance(self.scale, torch.Tensor) else self.scale
```

Comment:
Not sure I understand why we need previous `scale` to compute the scaling factor?

Reply:
maybe better to rename func to `update` instead of `compute`?